packages/design-system/src/components/TextField/LabelMask/LabelMask.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState, useRef } from 'react';
+
+export type LabelMaskMask = 'phone' | 'ssn' | 'date' | 'day_month';
+
+export interface LabelMaskProps {
+  /**
+   * Must contain a `TextField` component
+   */
+  children: React.ReactNode;
+  labelMask?: LabelMaskMask | ((rawInput: string) => string);
+  // placeholder?: string;
+  // value?: string;
+}
+
+const LabelMask = ({ children, labelMask }: LabelMaskProps) => {
+  const labelMaskPattern = {
+    phone: '[0-9-]*',
+    ssn: '[0-9-*]*',
+    date: '[0-9-]*',
+    day_month: '[0-9.,-]*',
+  };
+
+  /**
+   * Get the child text field. Called as a method so that
+   * updates to the field cause the mask to re-render
+   * @returns {React.ReactElement} Child TextField
+   */
+  const field = (): React.ReactElement => React.Children.only(children as React.ReactElement);
+  const initialValue = field().props.value || field().props.defaultValue || '### ### ###';
```

Comment:
This component won't support `defaultValue` unless we add that feature later. We'll always assume this is a controlled component.

Reply:
(Which also means this whole line will be gone)